.github/workflows/workflow-repo-standards-validation.yml

```diff
@@ -0,0 +1,120 @@
+name: Reusable workflow to validate organization standards
+
+on:
+  workflow_call:
+
+jobs:
+  validate-name:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check repository name
+      run: |
+        repo_name="${{ github.repository }}"
+        lower_repo_name="${repo_name,,}"
+        
+        if [[ "$repo_name" != "$lower_repo_name" ]]; then
+          echo "The repository name '$repo_name' should contain only lowercase letters." >&2
+          exit 1
+        elif [[ "$repo_name" == *"_"* ]]; then
+          echo "The repository name '$repo_name' should contain dashs ( - ) and not underscores ( _ )." >&2
+          exit 1
+        fi
+      if: always()
+      shell: bash
+
+    - name: Check repository visibility
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        repo_info=$(curl -H "Authorization: token $GITHUB_TOKEN" -s "https://api.github.com/repos/${{ github.repository }}")
+        repo_visibility=$(echo $repo_info | jq -r .private)
+        
+        if [[ "$repo_visibility" == "true" ]]; then
+          echo "The repository '${{ github.repository }}' is private and needs to be public". >&2
+          exit 1
+        fi
+      if: always()
+      shell: bash
+
+    - name: Check repository description
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        repo_info=$(curl -H "Authorization: token $GITHUB_TOKEN" -s "https://api.github.com/repos/${{ github.repository }}")
+        repo_description=$(echo $repo_info | jq -r .description)
+        
+        if [[ "$repo_description" == "null" || -z "$repo_description" ]]; then
+          echo "The repository does not have a description, please add one." >&2
+          exit 1
+        fi
+      if: always()
+      shell: bash
+
+    - name: Check branch protection
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        protection=$(curl -H "Authorization: token $GITHUB_TOKEN" -s "https://api.github.com/repos/${{ github.repository }}/branches/main/protection")
+        message=$(echo $protection | jq -r .message)
+
+        echo $protection | jq -r .message
+        
+        if [[ "$message" == "Branch not protected" ]]; then
+          echo "Error: The main branch is not protected!" >&2
+          exit 1
+        fi
+      if: always()
+      shell: bash
+
+    - name: Checkout code
+      uses: actions/checkout@v3
+
+    - name: Check for README.md
+      run: |
+        FILE="README.md"
```

Comment:
This already exist:

https://github.com/marketplace/actions/file-existence

and is an improvement on creating a step for each file.

Reply:
I just reused the code I already created from this comment you made
https://github.com/ai-cfia/github-workflows/pull/9#pullrequestreview-1660351229

do you want me to scrap the code and just use the reusable one you linked?

Reply:
I thought we discussed this verbally and I had made it clear we favor reuse? Not sure why this comes up again.

Reply:
OK, just to clarify: if I point to an existing piece of code that does the same, I expect you to use that piece of code. Every new line of code you create is an additional maintenance burden on the team.

Reply:
I ended up using Octokit for both